### PR TITLE
Add a section in Introduction with a mlr->mlr3 lookup table

### PR DESCRIPTION
**Pull request by mboecker**
This should make it easier for long-time mlr users to switch to mlr3.
Is this roughly what you wanted, or should I go more in-detail?
I could also extend the table if you feel like this is missing crucial elements of mlr or mlr3.

### CODE REVIEW

File: bookdown/01-introduction.Rmd [hunk 1]
```diff
@@ -22,3 +27,41 @@ Other _building blocks_ include:
 - Resampling methods
 
 These additional blocks are provided by our extension packages such as `r gh_pkg("mlr-org/mlr3tuning")`, `r gh_pkg("mlr-org/mlr3pipelines")` or `r gh_pkg("mlr-org/mlr3featuresel")`.ing.
+
+## Equivalent elements from `mlr`
```
Comment (pat-s) [inline bookdown/01-introduction.Rmd]: I would simply call it "transition table" or similar. 
Not everything will be equivalent in mlr3 and the list will also contain a lot of missing items.

File: bookdown/03-train-predict.Rmd [hunk 2]
```diff
@@ -18,11 +18,11 @@ Additionally, this chapter includes the following use-cases
 The first step is to generate the following `mlr3` objects from the [task dictionary](#tasks) and the [learner dictionary](#learners), respectively:
 
 1. The classification task
-    ```{r}
+    ```{r train-predict-task}
```
Comment (pat-s) [inline bookdown/03-train-predict.Rmd]: These changes were already merged in your prior PR right?

Comment (mboecker) [reply bookdown/03-train-predict.Rmd]: Yes exactly I merely merged the branch into this one before the PR was merged

**Comment by mboecker** [comment]
> The goal should be to have a table for every exported function in _mlr_.

Just to be clear, this means finding every function from https://mlr.mlr-org.com/reference/index.html in mlr3?

**Comment by pat-s** [comment]
> Just to be clear, this means finding every function from mlr.mlr-org.com/reference/index.html in mlr3?

Somewhat yes. Not every function is mirrored and some old ones are now wrapped into one mlr3 class.

**Comment by jakob-r** [comment]
I think the way it is programmed is a bit hard to read. First question, do we have/want A) a 1-1 relation, B) 1-n/n-1 or C) many to many in this table?

A) Do a named character vector `c("mlrfun" = "mlr3fun")` or vice versa. The table consists of `names(x)` and `x`.
B) Do a named list `list("mlrfun" = c("mlr3fun1", "mlr3fun2"))`. The table consists of `names(x)` and `lapply(x, paste, collapse = ", ")`. Of course you can make it more flexible here and make each list item a list with some description or whatever.
C) I havent made up my mind yet.

D) Do we even have to do this programmatically?

**Comment by pat-s** [comment]
> D) Do we even have to do this programmatically?

Not really but is it so much easier not to do it programmatically?

Always happy to see improved layouts of this approach.

**Comment by mboecker** [comment]
Whats the consensus here? I don't think the specific layout of the vectors is important because we dont expect users or anyone to use them without the book. For the book, I see the main usecase as searching (ctrl+F) for the item you want to have translated.

**Comment by jakob-r** [comment]
> I don't think the specific layout of the vectors is important because we dont expect users or anyone to use them without the book.

As a user of the code I think it is very complicated to write the mapping because I would have to count where to put the new mlr3 function so it relates to the correct mlr function.

**Comment by mboecker** [comment]
I agree. In this case, it would be my favorite option to go with n-m mapping using `c("a / b" = "c / d")`

**Comment by pat-s** [comment]
I agree with both of your points here. 
I can see why the n-m mapping is actually easier for writing.
However, aren't we running into an ugly alignment with this?

**Comment by jakob-r** [comment]
I don't get how you mean n-m mapping with `c("a / b" = "c / d")` but I am sure it will work.

I also don't now what the alignment issue should be :/

**Comment by pat-s** [comment]
@mboecker 	Can you add one or two lines showing how you exactly would do it so that we can have a look at the visual output?

**Comment by mboecker** [comment]
```r
t = c(
  "RegrTask / ClassifTask"         = "TaskRegr / TaskClassif",
  "makeRegrTask / makeClassifTask" = "TaskRegr$new() / TaskClassif$new()")
knitr::kable(names(t), t)
```
This is based on jakob's approach B. Output would look the same as before

**Comment by jakob-r** [comment]
I think the table should be more `pure`. Becasue we also have `ClusterTask`, `SurvTask` and many more. For everything that is not in pure mlr3 we can add eg. `mlr3surv::`
If you want categories order the rows somehow or use something more complicated:
```r
t = "
category , mlr             , mlr3              , note
Task     , makeClassifTask , TaskClassif$new() , NA
Task     , RegrTask        , TaskRegr          , Class
Task     , makeRegrTaskm   , TaskRegr$new()    , NA
"
t = read.table(text =  t, sep = ",", header = TRUE)
t = knitr::kable(t)
kableExtra::collapse_rows(t, columns = 1)
```

**Comment by pat-s** [comment]
2019-06-21-105113_screenshot

**Comment by mllg** [comment]
Note: You can use `mlr3misc::rowwise_table()` to write down human-readable tables in R (idea stolen from `tibble::tribble`).

See https://github.com/mlr-org/mlr3/blob/master/R/mlr_reflections.R#L67 for an example.

**Comment by mllg** [comment]
AFAICT `kableExtra` can only produce HTML tables. This is a road block if we want to compile the book into a PDF in the future. Is there an alternative package which can handle both output types?

**Comment by jakob-r** [comment]
> AFAICT `kableExtra` can only produce HTML tables. 

It can produce both and nearly all stylings (or whatever you want to call them) translate into html and latex.

**Comment by mllg** [comment]
Oh okay then.

**Comment by pat-s** [comment]
Merging this now so that we have a first draft that we can continue on.

Future PRs should be of smaller size.

@mb706 	@mllg 	
If you implement new features in the future, it would be great if you could also update this list.
I know, feels like a 2nd NEWS file, but I can't track every new feature implementation to update this list alone. 
There is also still a lot missing and I hope @mboecker 	can add some more entries in the near future :)

**Pull request merged**